### PR TITLE
Update examples to use async/await

### LINKED ISSUE #570: Update documentation
The [connection pool](https://aiomysql.readthedocs.io/en/latest/pool.html) documentation seems to still use the old `yield from` and `@asyncio.coroutine` syntax, instead of the async-await syntax. Please update.

### PR DESCRIPTION
**Pull request by anand2312**
Closes #570

This pull request updates the examples in the documentation to use `async/await` syntax instead of the old `@asyncio.coroutine` syntax.

### CODE REVIEW

**Comment by H4ckerxx44** [comment]
Why is this not merged yet? It's quite painful to have examples very out of date..

**Comment by anand2312** [comment]
@H4ckerxx44 
This project is dead :(

**Comment by H4ckerxx44** [comment]
That's quite sad, thanks for the response though!

File: docs/cursors.rst [hunk 1]
```diff
@@ -26,23 +26,22 @@ Cursor
 
         loop = asyncio.get_event_loop()
 
-        @asyncio.coroutine
         def test_example():
```
Comment (Nothing4You) [inline docs/cursors.rst:29]: this should be async now

File: docs/connection.rst [hunk 2]
```diff
@@ -23,18 +23,17 @@ Example::
 
     loop = asyncio.get_event_loop()
 
-    @asyncio.coroutine
-    def test_example():
-        conn = yield from aiomysql.connect(host='127.0.0.1', port=3306,
+    async def test_example():
+        conn = await aiomysql.connect(host='127.0.0.1', port=3306,
                                            user='root', password='', db='mysql',
```
Comment (Nothing4You) [inline docs/connection.rst:28]: indent no longer matches

File: docs/cursors.rst [hunk 3]
```diff
@@ -26,23 +26,22 @@ Cursor
...
-            conn = yield from aiomysql.connect(host='127.0.0.1', port=3306,
+            conn = await aiomysql.connect(host='127.0.0.1', port=3306,
                                                user='root', password='',
```
Comment (Nothing4You) [inline docs/cursors.rst:31]: indent no longer matches

File: docs/cursors.rst [hunk 4]
```diff
@@ -183,22 +182,22 @@ Cursor
         query using :meth:`Cursor.execute()` to get any OUT or INOUT values.
         Basic usage example::
 
-            conn = yield from aiomysql.connect(host='127.0.0.1', port=3306,
+            conn = await aiomysql.connect(host='127.0.0.1', port=3306,
                                                user='root', password='',
```
Comment (Nothing4You) [inline docs/cursors.rst:186]: indent no longer matches

File: docs/cursors.rst [hunk 5]
```diff
@@ -183,22 +182,22 @@ Cursor
...
                                                db='mysql', loop=self.loop)
 
-            cur = yield from conn.cursor()
-            yield from cur.execute("""CREATE PROCEDURE myinc(p1 INT)
+            cur = await conn.cursor()
+            await cur.execute("""CREATE PROCEDURE myinc(p1 INT)
                                    BEGIN
```
Comment (Nothing4You) [inline docs/cursors.rst:191]: indent no longer matches

File: docs/cursors.rst [hunk 6]
```diff
@@ -292,21 +291,20 @@ Cursor
 
         loop = asyncio.get_event_loop()
 
-        @asyncio.coroutine
-        def test_example():
-            conn = yield from aiomysql.connect(host='127.0.0.1', port=3306,
+        async def test_example():
+            conn = await aiomysql.connect(host='127.0.0.1', port=3306,
                                                user='root', password='',
```
Comment (Nothing4You) [inline docs/cursors.rst:296]: indent no longer matches

File: docs/cursors.rst [hunk 7]
```diff
@@ -332,21 +330,20 @@ Cursor
...
```
Comment (Nothing4You) [inline docs/cursors.rst:335]: indent no longer matches

File: docs/pool.rst [hunk 8]
```diff
@@ -14,20 +14,19 @@ The basic usage is::
 
     loop = asyncio.get_event_loop()
 
-    @asyncio.coroutine
-    def go():
-        pool = yield from aiomysql.create_pool(host='127.0.0.1', port=3306,
+    async def go():
+        pool = await aiomysql.create_pool(host='127.0.0.1', port=3306,
                                                user='root', password='',
```
Comment (Nothing4You) [inline docs/pool.rst:19]: indent no longer matches

**Review by Nothing4You — CHANGES_REQUESTED** [review]
thanks for the PR, I've left a few comments requesting changes, one they are applied I will merge this.

I haven't tested the updated examples but the changes look good.
I plan to do a proper review of the examples before the next release.

**Review by Nothing4You — APPROVED** [review]
implemented requested changes

**Pull request merged**